cmake/Intel.cmake

```diff
@@ -0,0 +1,92 @@
+
+set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fpp -fno-alias -auto -safe-cray-ptr -ftz -assume byterecl -nowarn -sox -align array64byte")
+set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -qno-opt-dynamic-align")
+
+
+if(32BIT)
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -i4 -real-size 32")
+    add_definitions(-DOVERLOAD_R4)
+    add_definitions(-DOVERLOAD_R8)
+else()
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -i4 -real-size 64")
+    if(NOT REPRO)
+        set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -no-prec-div -no-prec-sqrt")
+    endif()
+endif()
+
+if(REPRO)
+    add_definitions(-DREPRO)
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -qno-opt-dynamic-align")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -qno-opt-dynamic-align")
+else()
+    if(AVX2)
+        set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -xCORE-AVX2 -qno-opt-dynamic-align")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -xCORE-AVX2 -qno-opt-dynamic-align")
+    else()
+        set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -xHOST -qno-opt-dynamic-align")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -xHOST -qno-opt-dynamic-align")
+    endif()
+endif()
+
+if(REPRO)
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -O2 -debug minimal -fp-model consistent -qoverride-limits -g -traceback")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O2 -debug minimal")
+else()
+    if(DEBUG)
+        set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -g -O0 -check -check noarg_temp_created -check nopointer -warn -warn noerrors -fp-stack-check -fstack-protector-all -fpe0 -debug -traceback -ftrapuv")
```

Comment:
We lost our "-link_mpi=debug" flag again.

Reply:
Thanks!

https://github.com/ufs-community/ufs-weather-model/blob/develop/cmake/Intel.cmake is what I had used. I will update this version, but do we address it with the ufs-community/ufs-weather-model as well or i'm assuming somewhere there's a branch that addresses this?

Reply:
I think that for the uncoupled model it is perhaps not crucial for debugging. But for the coupled model it appears to make a big difference in creating error messages that correctly point to the component model where the debug options are triggering a fault. Our experience was that w/o this flag, the model would fault in some random location (usually the ice model).